src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -765,62 +765,153 @@ private async Task<bool> AreCentralVersionRequirementsSatisfiedAsync(RestoreRequ
                 return true;
             }
 
-            IEnumerable<LibraryDependency> dependenciesWithVersionOverride = restoreRequest.Project.TargetFrameworks.SelectMany(tfm => tfm.Dependencies.Where(d => !d.AutoReferenced && d.VersionOverride != null));
+            bool result = true;
 
-            if (restoreRequest.Project.RestoreMetadata.CentralPackageVersionOverrideDisabled)
+            if (!restoreRequest.PackageSourceMapping.IsEnabled && httpSourcesCount > 1)
+            {
+                // Log a warning if there are more than one configured HTTP source and package source mapping is not enabled
```

Comment:
```suggestion
                // Log a warning if there are more than one configured HTTPS source and package source mapping is not enabled
```

Reply:
That way there's no confusion with comments elsewhere about raising warnings/errors for http usage instead of https.